Code/Legacy/CryCommon/BaseTypes.h

```diff
@@ -11,12 +11,9 @@
 #define CRYINCLUDE_CRYCOMMON_BASETYPES_H
 #pragma once
 
-#include "CompileTimeAssert.h"
-
-
-COMPILE_TIME_ASSERT(sizeof(char) == 1);
-COMPILE_TIME_ASSERT(sizeof(float) == 4);
-COMPILE_TIME_ASSERT(sizeof(int) >= 4);
+static_assert(sizeof(char) == 1);
+static_assert(sizeof(float) == 4);
+static_assert(sizeof(int) >= 4);
```

Comment:
For these static asserts, I have no clue which platforms they are trying to protect against.
They don't seem useful atll

Reply:
no idea, my guess is that there may be some other operations around that can fail if this is not the case (e.g. in `BitFiddling.h`)